### ISSUE TITLE
Auto welcome to new followers

### ISSUE THREAD

**Issue by t4coaching** [igdmapps/igdm-pro-release#76]
Auto welcome to new followers not working guys?

I definitely have it set up correctly and the app is open and running whether the laptop is on or off. 

Been trying it for 2 days now and not one message has been sent. 

my account is @trans4ormcoaching

Many thanks

Dave

**Comment by t4coaching** [comment]
PS it's a brand new Samsung laptop on windows 11 on Chrome

**Comment by ifedapoolarewaju** [comment]
Hi @t4coaching , sorry to hear that you are facing challenges with using this feature.

- When you say the "App is open and running", are you referring to the IGdm Pro app?
- About how many new followers have you had since when you enabled the feature?

**Comment by t4coaching** [comment]
Yes the IGdm Pro app is saved to my task bar and open.  Please just clarify what open/running actually means

I've had 30 new followers in the last 24 hours so prob 50 total since I set up IGdm pro

Thanks

**Comment by ifedapoolarewaju** [comment]
> Yes the IGdm Pro app is saved to my task bar and open. Please just clarify what open/running actually means

Sure, keeping it open just means that the app window is not closed, but is instead minimised.
By minimise, I mean the button on the extreme left as in the image below

![image](https://github.com/user-attachments/assets/f6504bac-0c4f-4092-9bed-7952543cb4f5)

**Comment by ifedapoolarewaju** [comment]
Are you currently on free trial, or do you have an active IGdm Pro License? Of course it should work while you're on free trial, but I'm just asking to get more clarity on how to investigate

**Comment by t4coaching** [comment]
free atm

**Comment by t4coaching** [comment]
Yeah I have in minimised all the time. I've just set it to run as admin and it's sent a message

**Comment by ifedapoolarewaju** [comment]
Hmm ... so now it's working for you?

**Comment by t4coaching** [comment]
It's sent one message yes i'll keep an eye

**Comment by ifedapoolarewaju** [comment]
Yes please let me know how this progresses.

**Comment by t4coaching** [comment]
It's working now. The fix was that I had to search the app from the taskbar and right-click the app icon. Then there is an option to 'run as administrator'.  As soon as that's been selected, a notification pops up that the new follower automated welcome message is active.

**Comment by t4coaching** [comment]
It's sketchy again now. It's only sent 1 more message and it sent it twice to the same person? If the app is left running and I close the lid on the laptop wih it minimised and still running it should still work right? The laptop doesn't need to be open 24/7 for it to work?

**Comment by t4coaching** [comment]
and now sent one to an older follower

**Comment by ifedapoolarewaju** [comment]
> If the app is left running and I close the lid on the laptop wih it minimised and still running it should still work right? The laptop doesn't need to be open 24/7 for it to work?

It depends on if your laptop goes to sleep when you close the lid. If your laptop doesn't go to sleep, and continues to have internet connection, it should work. To be sure, you could leave it open instead, however, if you are certain your laptop does not go to sleep when you close the lid, you can close it.

**Comment by ifedapoolarewaju** [comment]
> and now sent one to an older follower

This is unusual, has it missed out on some followers now? I mean have you gotten new followers where it didn't send to them since it started working?

**Comment by t4coaching** [comment]
Don't think so, but it's messaged two old followers

**Comment by t4coaching** [comment]
I've also got friends to unfollow and re-follow and it doesn't count them as new followers or message them.

**Comment by ifedapoolarewaju** [comment]
In the setting, how long is the period you set for it to send followers messages? 30 minutes?

**Comment by t4coaching** [comment]
yes I did 30 mins just to test it as quickly as possible

**Comment by ifedapoolarewaju** [comment]
Ok, so in order to test it when your friends unfollow you, you need to relaunch the app (close it this time, and then open it again), then have them follow you again, and wait 30 minutes. It ought to send the message in that case.

Otherwise, the app won't know that those users had previously unfollowed you.

**Comment by t4coaching** [comment]
Hi again guys.

11 new followers again since my last post yesterday,  and it's now not working again at all. No new messages sent to any new followers.

Any ideas?

Thanks

**Comment by ifedapoolarewaju** [comment]
Hi @t4coaching I'm trying to investigate what's happening for your case specifically because it seems unusual.

Could you please re-start the IGdm Pro app, and send post a message here immediately after you do? That way I can use that time as a starting point to investigate log analytics on my end.

**Comment by t4coaching** [comment]
Done at 11.27 am and the conformation pop up appeared to stating that the message is active

**Comment by ifedapoolarewaju** [comment]
Thank you. I'll investigate and share my findings.

Thank you for your patience.

**Comment by t4coaching** [comment]
Thanks so much

**Comment by ifedapoolarewaju** [comment]
hi @t4coaching so I'm noticing the actual problem is that when the app tries to send a message, it fails. Are you able to send a message using the IGdm Pro app? So like if you try to send the message to one of your new followers manually using the IGdm Pro app, what happens?

Also, it seems whenever you launch the app, it launches multiple windows at once? This may be the reason messages are failing to send, so could you try avoiding this by only double-clicking the app icon once when you want to launch it?

**Comment by t4coaching** [comment]
I have noticed the multiple windows yes. It does it sometimes with a single click. I only really ever have one open as I close the others if it happens so there is usually just one running in the background when the laptop is closed. I have just sent test messages to my wife and daughter and nothing will send. It was working fine with manual messaging when I first downloaded it though so that's a new issue. The unsent messages have a clock icon as opposed to a tick.

**Comment by ifedapoolarewaju** [comment]
> I have just sent test messages to my wife and daughter and nothing will send

Yes, this is the issue that to auto-message is also facing in the background, and I suspect it's due to the multiple windows launching. Even though you close one of them afterwards, I think it may already have had some undesirable side effect at that point.

I'm trying to see if I can roll out an update that prevents it from launching multiple windows, but if you could try avoiding it, that may help. Perhaps launch it with a "right click", or any other option that may be available on your machine.

**Comment by t4coaching** [comment]
Yes it will launch with a right click from the minimised state. Any further updates?

**Comment by ifedapoolarewaju** [comment]
Hello, apologies for the late update on this.

A new version (v3.3.5) has been released that fixes this problem by automatically detecting when the messages stop sending, and it restarts the app by itself.

**Comment by t4coaching** [comment]
Unfortunately iG messaged me and threatened me with my account being closed
if I continued to use the new follower software

**Comment by ifedapoolarewaju** [comment]
sorry to hear about this @t4coaching.

I can understand if you choose to discontinue your usage of the IGdm Pro app. However, I have recently released a new version v3.3.6 that reduces the chances of you receiving these warnings. This is just an FYI, if you are still looking to give it a try, it's available for download on [the website](https://pro.igdm.me)

**Comment by t4coaching** [comment]
I'll try it again

**Comment by JasonBroderick** [comment]
Hey mate, I'm testing 3.3.7 (free trial) and having a similar issue. I have auto dm new followers, 30 minutes, toggle active, tested manual messages working from the app, app is open and active at all times and my test account gets ~200 new followers a day on average. 

So far the app has only sent one message, so I relaunched as administrator to make sure, and it sent 1 more message (now two total) but since then nothing again. 

I think it would be very helpful to show a Log somewhere that we can see, that shows new followers recognised, messages attempted, and a status (successful, failed, delayed etc) 

That way I could see if it was maybe not registering new followers, or not sending the messages, or help diagnose the issue :)
